From 520c661f4357d49b57f11638ac8c913ca0996a6a Mon Sep 17 00:00:00 2001
From: Ryan McCabe <rmccabe@redhat.com>
Date: Wed, 4 Mar 2015 08:05:24 -0500
Subject: [PATCH] luci: Add support for the fence_apc cmd_prompt attr

Add support for setting/viewing the newly-added "cmd_prompt"
attribute for fence_apc fence devices.

Resolves: rhbz#1010400

Signed-off-by: Ryan McCabe <rmccabe@redhat.com>
---
 luci/templates/fence_devices.html | 7 +++++++
 luci/validation/validate_fence.py | 1 +
 2 files changed, 8 insertions(+)

diff --git a/luci/templates/fence_devices.html b/luci/templates/fence_devices.html
index f7b154c..3417e64 100644
--- a/luci/templates/fence_devices.html
+++ b/luci/templates/fence_devices.html
@@ -54,6 +54,13 @@
       </td>
     </tr>
     <tr>
+      <td>Force Command Prompt</td>
+      <td>
+        <input type="text" class="text" name="cmd_prompt"
+          py:attrs="cur_fencedev and {'value': cur_fencedev.getAttribute('cmd_prompt')} or {}"/>
+      </td>
+    </tr>
+    <tr>
       <td>Power Wait (seconds)</td>
       <td>
         <input type="text" class="text" name="power_wait"
diff --git a/luci/validation/validate_fence.py b/luci/validation/validate_fence.py
index 04996fa..9f43848 100644
--- a/luci/validation/validate_fence.py
+++ b/luci/validation/validate_fence.py
@@ -149,6 +149,7 @@ def val_apc_fd(fencedev, fence_name, **kw):
 		('passwd', False),
 		('passwd_script', False),
 		('power_wait', False),
+		('cmd_prompt', False),
 		('power_timeout', False),
 		('shell_timeout', False),
 		('login_timeout', False),
-- 
2.1.0

